src/metabase/util/schema.clj

```diff
@@ -330,15 +330,22 @@
 
 (def Parameter
   "Schema for a valid Parameter."
-  (with-api-error-message {:id       NonBlankString
-                           s/Keyword s/Any}
-    (deferred-tru "parameter must be a map with String :id key")))
+  (with-api-error-message {:id                       NonBlankString
+                           :type                     NonBlankString
+                           ;; Allow blank name and slug #15279
+                           (s/optional-key :name)    s/Str
+                           (s/optional-key :slug)    s/Str
+                           (s/optional-key :default) s/Any
+                           s/Keyword                 s/Any}
```

Comment:
I wonder how risky it would be to make this a closed schema, restricted to the expected keys. That feels like what @camsaul is requesting on #23208 -- validation that ensures that we don't persist garbage keys for whatever reason.

But it would be an issue if there's already garbage keys saved in the DB. Maybe it's fine if we only enforce the schema on writes, not reads.

Reply:
I think it's risky to enforce a closed schema on writes and not reads. 
Imagine someone updating an existing parameter with unallowed keys, FE will probably send the parameter with unallowed keys and we'll deny it.

This could be done if FE has a filter to send only allowed keys, but I don't think the added complexity will add much value here.